src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -626,12 +626,11 @@
     (.setHandlers hc (into-array Handler [(:handlers webserver-context)]))
     (let [shutdown-timeout (when (not (nil? (:shutdown-timeout-seconds options)))
                              (* 1000 (:shutdown-timeout-seconds options)))
-          maybe-zipped (if (or (not (contains? options :gzip-enable))
-                               (:gzip-enable options))
+          maybe-zipped (if (:gzip-enable options true)
                          (gzip-handler hc)
                          hc)
           maybe-logged (if log-handler
-                         (doto log-handler (.setHandler hc))
+                         (doto log-handler (.setHandler maybe-zipped))
```

Comment:
This was the real problem.  We weren't passing in the handler layered with the `GzipHandler` when the log-handler was configured.